Modules/fact_manager.py

```diff
@@ -0,0 +1,337 @@
+
+        Returns: Nothing
+        """
+        if not await self.exists(name, lang):
+            log.exception("Attempted edit on non-existent fact.")
+            raise ValueError
+
+        try:
+            current_fact = await self.find(name, lang)
+
+            edit_query = sql.SQL(f"UPDATE {self._FACT_TABLE} SET message=%(message)s, "
+                                 f"edited=%(edit_time)s WHERE name=%(name)s AND lang=%(lang)s")
+
+            query_values = {"edit_time": datetime.datetime.now(datetime.timezone.utc),
+                            "message": new_message, "name": name, "lang": lang}
+
+            await self.query(edit_query, query_values)
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            log.exception(f"Editing fact '{name}-{lang}' failed.")
+            raise error
+        else:
+            await self.log(name, lang, editor, 'Edited', new_message, current_fact.message)
+
+    async def exists(self, name: str, lang: str) -> bool:
+        """
+        Check if a fact exists, without having to substantiate a full Fact object.
+
+        >>> fact_existence = self.exists('test', 'en')
+
+        Args:
+            name: name of fact
+            lang: language ID of fact
+
+        Returns: True/False, if already exists.
+
+        """
+        query = sql.SQL(f"SELECT COUNT(*) message FROM "
+                        f"{self._FACT_TABLE} WHERE name=%s AND lang=%s")
+
+        try:
+            result = await self.query(query, (name, lang))
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # Check for offline database, or issues with the query.
+            log.exception(f"Could not establish existence of {name}-{lang}")
+            raise
+
+        # We are only getting a single integer as a response, so we can unpack it by index.
+        # it will always return a single integer.
+        return result[0][0]
+
+    async def fact_history(self, fact_name: str, fact_lang: str) -> list:
+        """
+        Pulls the last 5 transaction logs for a fact, as a tuple
+
+        >>> history_list = self.fact_history("test", "en")
+
+        Args:
+            fact_name: Name of the fact
+            fact_lang: LangID of the fact.
+
+        Returns: tuple of transaction log items, for fact.
+        """
+        query = sql.SQL(f"SELECT name, lang, author, message, ts, old, new "
+                        f"FROM {self._FACT_LOG} WHERE name=%s AND lang=%s "
+                        f"ORDER BY ts DESC LIMIT 5")
+
+        query_data = (fact_name, fact_lang)
+
+        try:
+            result = await self.query(query, query_data)
+        except (psycopg2.ProgrammingError, psycopg2.DatabaseError) as error:
+            # ProgrammingError is a query failure, DatabaseError is database unavailable.
+            log.exception(f"Unable to retrieve history for {fact_name}-{fact_lang}")
+            raise
```

Comment:
While valid, we should re-raise the exception explicitly as to not confuse developers not familiar with this syntax.
```suggestion
            raise error
```

Reply:
Every instance of a bare raise has been replaced with ``raise error``

Fixed in 0d99f5d9.